File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/Revision.java
package com.atlaskv.sdk.models;

/**
 * Immutable representation of a single historical key revision.
 *
 * @param revisionNumber version revision number
 * @param value          value of the key at this revision (null if deleted/expired)
 * @param timestamp      epoch timestamp of this revision
 * @param operation      operation type (PUT, DELETE, EXPIRE, ROLLBACK)
 * @param nodeId         ID of the node that committed the revision
 * @param leaseId        associated lease ID (null if none)
 * @param ttl            associated TTL string (null if none)
 */
public record Revision(
        long revisionNumber,
        String value,
        long timestamp,
        String operation,
        String nodeId,
        String leaseId,
        String ttl
) {}

File: atlaskv-ts-sdk/src/models/Revision.ts
/**
 * Represents a single historical revision of a key-value pair.
 */
export interface Revision {
  /** The version number of this revision. */
  revisionNumber: number;
  /** The value of the key at this revision (null if deleted or expired). */
  value: string | null;
  /** The epoch timestamp in milliseconds when the revision was committed. */
  timestamp: number;
  /** The operation type that created this revision (e.g., PUT, DELETE, EXPIRE, ROLLBACK). */
  operation: string;
  /** The ID of the node that committed the revision. */
  nodeId: string;
  /** The lease ID associated with this revision, or null if none. */
  leaseId: string | null;
  /** The TTL duration string associated with this revision, or null if none. */
  ttl: string | null;
}
